nixos/modules/services/hardware/thermald.nix

```diff
@@ -23,6 +23,15 @@ in {
         default = null;
         description = "the thermald manual configuration file.";
       };
+
+      adaptive = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable adaptive mode, only working on kernel versions greater than 5.8.
+          Thermald will detect this itself, safe to enable on kernel versions below 5.8.
```

Comment:
Should this be enabled by default, if it's safe even if not supported by the kernel?

Reply:
Two reasons for not enabling this by default:
1. Thermald doesn't enable it by default, so we shouldn't either to not surprise existing users with new behavior
2. There are some existing platforms where the adaptive doesn't work correctly, so enabling this should be a user decision, see https://github.com/intel/thermal_daemon/issues?q=is%3Aissue+is%3Aopen+adaptive, in case of issues the user can easily rollback this config option.